### PR TITLE
task(timing): Modifies timing semantics to mimic prior timings

### PR DESCRIPTION
**Pull request by scalvert**
This is the last (hopefully!) of a series of simplifications to this addon. In this last change, we've slightly modified the timing semantics to attempt to mirror the succession of code executions that culminates with the user thennables running. We've added a wrapping `Promise` to ensure that the chaining of the top level deferred more closely resembles the prior timings.

### CODE REVIEW

**Comment by scalvert** [comment]
@rwjblue this is the PR for the work we paired on Friday. I managed to fix the tests (there was a bug with `routeSettled` where we were returning the deferred rather than the `whenRouteIdle` promise.

It should be gtg.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { beginTransition, endTransition, routeSettled, setupRouter, reset, didTransition, whenRoutePainted, whenRouteIdle, } from 'ember-app-scheduler/scheduler';
```
Comment (rwjblue) [inline index.d.ts]: Hmm, I don't think these .d.ts files should be committed

File: scheduler.d.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { Promise } from 'rsvp';
```
Comment (rwjblue) [inline scheduler.d.ts]: ✂️

File: tests/unit/scheduler-test.js [hunk 3]
```diff
@@ -33,18 +33,23 @@ module('Unit | Scheduler', function(hooks) {
     assert.expect(3);
 
     beginTransition();
+    console.log('beginning first transition');
```
Comment (rwjblue) [inline tests/unit/scheduler-test.js]: Do you mean to leave these `console.log`?

**Comment by scalvert** [comment]
Confirmed with @rwjblue offline that he's ok with this merging.

**Pull request merged**